sdk/storage/azure-storage-blob/azure/storage/blob/_upload_helpers.py

```diff
@@ -146,7 +152,9 @@ def upload_block_blob(  # pylint: disable=too-many-locals, too-many-statements
                     # Adjust total_size for encryption V2
                     total_size = adjusted_count
                     # V2 wraps the data stream with an encryption stream
-                    stream = GCMBlobEncryptionStream(cek, stream)
+                    if cek is None:
+                        raise ValueError("Generate encryption metadata failed. 'cek' is None.")
+                    stream = GCMBlobEncryptionStream(cek, stream)  # type: ignore
```

Comment:
`azure\storage\blob\_upload_helpers.py:157: error: Incompatible types in assignment (expression has type "GCMBlobEncryptionStream", variable has type "IO[Any]")  [assignment]`

Reply:
Can you try changing the type of `data_stream` in the `GCMBlobEncryptionStream` constructor to `IO[bytes]` instead of `BinaryIO` to see if that helps the second error? Will still need to ignore the first but that seems like a good fix for the second.